integration_tests/src/main/python/cast_test.py

```diff
@@ -304,7 +304,22 @@ def test_cast_array_to_string(data_gen, legacy):
     _assert_cast_to_string_equal(
         data_gen,
         {"spark.sql.legacy.castComplexTypesToString.enabled": legacy})
-
+    
+def test_cast_float_to_string():
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: unary_op_df(spark, FloatGen()).selectExpr("cast(cast(a as string) as float)"),
+        conf = {"spark.rapids.sql.castStringToFloat.enabled": True,
+                "spark.rapids.sql.castFloatToString.enabled": True})
+
+def test_cast_double_to_string():
+    conf = {"spark.rapids.sql.castFloatToString.enabled": True}
+    cast_func = lambda spark: unary_op_df(spark, DoubleGen()).selectExpr("cast(a as string)").collect()
+    from_cpu = with_cpu_session(cast_func, conf)
+    from_gpu = with_gpu_session(cast_func, conf)
+    cast_to_float_func = lambda row: row.a if row.a is None or row.a == 'NaN' else float(row.a)
+    from_cpu_float = list(map(cast_to_float_func, from_cpu))
+    from_gpu_float = list(map(cast_to_float_func, from_gpu))
+    assert from_cpu_float == from_gpu_float
```

Comment:
But NaN != NaN, so how will from_cpu_float ever be equal to from_gpu_float if there is a NaN in them?

Reply:
Okay so we are mixing types within an array. That is rather confusing but I guess it is okay.